### PR TITLE
Clean up config

### PR DESCRIPTION
**Pull request by JosephMarinier**
## Description:

This doesn't affect the app, it simply cleans up the code, making it shorter, while providing more robust typing for fields that can be `null` and fields with literal strings (select menus). Please see the commit descriptions.

I reduced the code by almost 100 lines! Splitting into different files added back 20 lines of imports and overhead (which is fine, those lines are trivial), and the extra typing added a bit more (but added great type security too), hence why the net reduction is 56 lines.

## Checklist:

You should check all boxes before the PR is ready. If a box does not apply, check it to acknowledge it.

* [x] **ISSUE NUMBER.** You linked the issue number (Ex: Resolve #XXX).
* [x] **PRE-COMMIT.** You ran pre-commit on all commits, or else, you
  ran `pre-commit run --all-files` at the end.
* [x] **USER CHANGES.** The changes are added to CHANGELOG.md and the documentation, if they impact
  our users.
* [x] **DEV CHANGES.**
    * Update the documentation if this PR changes how to develop/launch on the app.
    * Update the `README` files and our wiki for any big design decisions, if relevant.
    * Add unit tests, docstrings, typing and comments for complex sections.

### CODE REVIEW

File: webapp/src/pages/Settings.tsx [hunk 1]
```diff
@@ -914,7 +914,7 @@ const Settings: React.FC<Props> = ({ open, onClose }) => {
               key={field}
               label={field}
               value={value}
-              disabled={!resultingConfig[config] || isUpdatingConfig}
+              disabled={resultingConfig[config] === null || isUpdatingConfig}
```
Comment (christyler3030) [inline webapp/src/pages/Settings.tsx:728]: Could be extracted for reuse instead of checking every time.

File: webapp/src/pages/Settings.tsx [hunk 2]
```diff
@@ -135,17 +135,25 @@ const FIELD_COMMON_PROPS = {
   InputLabelProps: { shrink: true },
 } as const;
 
-const StringField: React.FC<
-  Omit<TextFieldProps, "onChange"> & FieldProps<string>
-> = ({ onChange, ...props }) => (
+const StringField = <T extends string | null>({
+  value,
```
Comment (christyler3030) [inline webapp/src/pages/Settings.tsx]: Out of curiosity: why is this explicit now?

**Review by christyler3030 — APPROVED** [review]

File: webapp/src/pages/Settings.tsx (hunk 2 above)
Comment (JosephMarinier) [reply webapp/src/pages/Settings.tsx]: To support `null` as a valid value, I now do `value={value ?? ""}`. Without that, `value={null}` would make the component "uncontrolled", and changing the `TextField` from controlled to uncontrolled (and vice versa) results in errors in the console. The `value={value ?? ""}` treats `null` like an empty string (which is a valid value for a controlled component), which makes the `placeholder: "null"` appear in pale gray.
You might notice that it is therefore impossible to set empty string to a nullable fields. We think this is good enough for now, as none of the nullable fields support the empty string as a valid value.
Does that make sense?

File: webapp/src/pages/Settings.tsx (hunk 1 above)
Comment (JosephMarinier) [reply webapp/src/pages/Settings.tsx:733]: Right. I hesitated. The pattern is repeated 9 times, but it's with 3 different things on the right, so it's not trivial. I have some ideas to go further in removing duplication, so I'll keep that in mind for future refactors.

**Pull request merged**